pkg/table/column.go

```diff
@@ -624,22 +642,19 @@ func getColDefaultValue(ctx expression.BuildContext, col *model.ColumnInfo, defa
 }
 
 func getColDefaultValueFromNil(ctx expression.BuildContext, col *model.ColumnInfo, args *getColOriginDefaultValue) (types.Datum, error) {
-	if !mysql.HasNotNullFlag(col.GetFlag()) && !mysql.HasNoDefaultValueFlag(col.GetFlag()) {
```

Comment:
Will this change affect reading? Because the PR handles the insert operation, the read operation lacks the judgment logic.

Reply:
Yes. My goal is to make `HasNoDefaultValueFlag` not affect the reading. I think `HasNoDefaultValueFlag` check should only check when insert, and not check when read operation.

`HasNoDefaultValueFlag` judgment logic here is introduced by #35072, and with this judgment logic, will cause bug in issue #51324 .